docs/01-app/05-api-reference/05-config/01-next-config-js/taint.mdx

```diff
@@ -0,0 +1,237 @@
+---
+title: taint
+description: Enable tainting Objects and Values.
+version: experimental
+---
+
+## Usage
+
+Experimental support for tainting Objects and Values.
+
+This flag enables usage of these React features:
+
+- [`experimental_taintObjectReference`](https://react.dev/reference/react/experimental_taintObjectReference)
+- [`experimental_taintUniqueValue`](https://react.dev/reference/react/experimental_taintUniqueValue)
+
+> **Good to know**: Activating this flag, enables the React `experimental` channel for `app` directory
+
+```ts filename="next.config.ts" switcher
+import type { NextConfig } from 'next'
+
+const nextConfig: NextConfig = {
+  experimental: {
+    ppr: 'incremental',
+  },
+}
+
+export default nextConfig
+```
+
+```js filename="next.config.js" switcher
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  experimental: {
+    ppr: 'incremental',
+  },
+}
+
+module.exports = nextConfig
+```
+
+The taint API includes two experimental functions, that allow you to mark objects and values, to prevent them from ever reaching the Client.
+
+- To taint objects references, use `experimental_taintObjectReference`.
+- To taint unique values, use `experimental_taintUniqueValue`.
+
+> **Warning:** Do not rely on the taint API, as your only mechanism to prevent exposing sensitive data to the client.
+
+It is recommended to model your data and APIs, so that sensitive data is not returned to contexts where it is not needed.
+
+For cases where:
+
+- The methods to read data are out of your control
+- You have to work sensitive data shapes not defined by you
+- Sensitive data has to be used while rendering your server components
+
+The taint API, allows you to be defensive, by declaratively and explicitly marking data that is not allowed to pass through the Server-Client boundary.
+
+When an object or value, is passed through the Server-Client boundary, React throws an error.
```

Comment:
Rearranged the sentences to introduce how it works first, wdyt? 

```suggestion
The taint APIs allows you to be defensive, by declaratively and explicitly marking data that is not allowed to pass through the Server-Client boundary. When an object or value, is passed through the Server-Client boundary, React throws an error.

This is helpful for cases where: 

- The methods to read data are out of your control
- You have to work with sensitive data shapes not defined by you
- Sensitive data is accessed during Server Component rendering

It is recommended to model your data and APIs so that sensitive data is not returned to contexts where it is not needed.
```